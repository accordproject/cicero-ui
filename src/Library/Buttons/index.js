/* React */
import React from 'react';
import PropTypes from 'prop-types';

/* Styling */
import styled from 'styled-components';
import { Button } from 'semantic-ui-react';

const ActionButton = styled.button.attrs({
  'aria-label': 'Upload Button',
})`
  position: relative;
  font-weight: 300;
  text-align: right;
  text-decoration: none;
  font-size: 0.87em;
  color: #76777D;
  background-color: inherit;
  border: 0;
  cursor: pointer;
  align-self: flex-start;
  &:hover {
    color: #0361DE;
    text-decoration: underline;
  }
  &:focus {
    outline: none;
    color: #3089FF;
    text-decoration: underline;
  }
  &:active {
    color: #3089FF;
    text-decoration: underline;
  }
`;

export const ImportComponent = props => {
  if (!props.onImportItem) return null;

  return(
    <ActionButton
      onClick={props.onImportItem}
      aria-label='Upload Button'
      className='cicero-ui__library-import-button'
    >
      Import from VS Code
    </ActionButton>
  );
}

export const UploadComponent = props => {
  if (!props.onUploadItem) return null;

  return(
    <ActionButton
      onClick={props.onUploadItem}
      aria-label='Import Button'
      className='cicero-ui__library-upload-button'
    >
      Add a new library item
    </ActionButton>
  );
}

export const NewItemComponent = props => {
  if (!props.onAddItem) return null;

  return(
    <Button
      content="New library Item"
      aria-label='Add Library Item Button'
      color="blue"
      fluid
      icon="plus"
      onClick={props.onAddItem}
      className='cicero-ui__library-add-item-button'
      style={{ margin: '5px 0' }}
    />
  );
}

ImportComponent.propTypes = { onImportItem: PropTypes.func };
UploadComponent.propTypes = { onUploadItem: PropTypes.func };
NewItemComponent.propTypes = { onAddItem: PropTypes.func };
