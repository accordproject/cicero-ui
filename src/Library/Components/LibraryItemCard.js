import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card, Image } from 'semantic-ui-react';

import CardActions from './CardActions';

const cardStyle = {
  margin: '10px 0',
  border: '1px solid #7B8084',
  borderRadius: '6px',
  backgroundColor: '#fff',
  textAlign: 'left',
  boxShadow: '0 1px 9px 0 rgba(0,0,0,0.1)',
}

const itemLogoStyle = {
  position: 'absolute',
  top: '13px',
  right: '16px',
  maxHeight: '23px',
}

const DescriptionContainer = styled(Card.Description)`
`;

/**
 * A Library Item Card component that displays each library item and it's details.
 */
const LibraryItemCard = props => (
  <Card
    fluid
    key={props.item.uri}
    style={cardStyle}
    className={`cicero-ui__library-card ${props.item.itemType}`}
  >
    <Card.Content className='cicero-ui__library-card-content'>
      <Image
        style={itemLogoStyle}
        src={props.item.logoUrl}
        className='cicero-ui__library-card-logo'
      />
      <Card.Header className='cicero-ui__library-card-header'>
        { props.item.displayName || props.item.name }
      </Card.Header>
      <Card.Meta className='cicero-ui__library-card-meta'>
        <span className='cicero-ui__library-card-item-type'>
          {props.itemTypeName}
        </span>
        &nbsp;|&nbsp;
        <span className='cicero-ui__library-card-item-version'>
          Version {props.item.version}
        </span>
      </Card.Meta>
      <DescriptionContainer>
        {props.item.description}
      </DescriptionContainer>
    </Card.Content>
    <CardActions
      item={props.item}
      onPrimaryButtonClick={props.onPrimaryButtonClick}
      onSecondaryButtonClick={props.onSecondaryButtonClick}
    />
  </Card>
);

LibraryItemCard.propTypes = {
  item: PropTypes.object.isRequired,
  onPrimaryButtonClick: PropTypes.func.isRequired,
  onSecondaryButtonClick: PropTypes.func.isRequired,
};

export default LibraryItemCard;
