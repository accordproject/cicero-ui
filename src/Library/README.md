## Cicero-UI - Library

### Usage

```shell
npm install @accordproject/cicero-ui
```

```js
import { Library } from '@accordproject/cicero-ui';

const LibraryComponent = props => (
    <Library
        onUploadItem={props.onUploadItem}
        onImportItem={props.onImportItem}
        onAddItem={props.onAddItem}
        items={props.items}
        onPrimaryButtonClick={props.onPrimaryButtonClick}
        onSecondaryButtonClick={props.onSecondaryButtonClick}
        itemTypes={props.itemTypes}
    />
);
```

### Props

- `items` [REQUIRED] : An `array` which contains library item objects. Required keys are: `uri`, `name`, `version` and `description`.
Optional keys are: `displayName`, `logoUrl` and `itemType` (for heterogeneous libraries)
- `onPrimaryButtonClick` [REQUIRED] : A `function` callback which is called when a primary button in a library item card is clicked. A single argument is passed - the item object.
- `onSecondaryButtonClick` [REQUIRED] : A `function` callback which is called when a secondary button in a library item card is clicked. A single argument is passed - the item object.
- `onUploadItem` [OPTIONAL] : A `function` callback which is called when a link to upload a new library item is clicked.
- `onImportItem` [OPTIONAL] : A `function` callback which is called when a link to import a new library item is clicked.
- `onAddItem` [OPTIONAL] : A `function` callback which is called when a link to add a new library item is clicked.
- `itemTypes` [OPTIONAL] : An `array` of item types, default: `[{ name: 'CLAUSE TEMPLATE', type: 'template', filterName: 'Templates' }]`.
`name` field is shown on the second line of the library card.
`type` field corresponds to `item.itemType` attribute and is used to match a given `item` to its itemType in the `itemTypes` array.
`filterName` field is shown as a label for the type filter checkbox. This filter is ommited if the library is homogeneous.

### Custom styling

The component provides classes which can be used to apply custom styles to its individual parts.

- `cicero-ui__library-search-input` : search input.
- `cicero-ui__library-card` : library card. An additional class of `item.itemType` will be added to the card,
so different CSS selectors can be used based on the item's `itemType` value.
- `cicero-ui__library-card-content` : content element inside a library item card.
- `cicero-ui__library-card-logo` : logo element inside a library item card.
- `cicero-ui__library-card-header` : header element inside a library item card.
- `cicero-ui__library-card-meta` : meta element inside a library item card.
- `cicero-ui__library-card-type` : item type name element inside a library item card.
- `cicero-ui__library-card-item-version` : version element inside a library item card.
- `cicero-ui__library-card-actions` : actions element inside a library item card.
- `cicero-ui__library-card-primary-btn` : primary button inside a library item card.
- `cicero-ui__library-card-secondary-btn` : secondary button inside a library item card.
- `cicero-ui__library-add-item-button` : add new library item button.
- `cicero-ui__library-upload-button` : upload new library item button.
- `cicero-ui__library-import-button` : import new library item button.
