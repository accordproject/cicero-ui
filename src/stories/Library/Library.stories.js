import React from "react";
import docs from "../../Library/README.md";
import { withA11y } from "@storybook/addon-a11y";
import { withKnobs, text } from "@storybook/addon-knobs";
import Library from "../../Library";

export default { title: "'Components/Library" };
export const library = () => {
  const items = [
    {
      uri: text(
        "First template's Uri",
        "https://templates.accordproject.org/archives/acceptance-of-delivery@0.13.1.cta"
      ),
      name: text("First template's Name", "Acceptance of Delivery"),
      version: text("First template's Version", "^0.20.10"),
      description: text(
        "First template's Description",
        "This clause allows the receiver of goods to inspect them for a given time period after delivery."
      ),
      itemType: 'template'
    },
    {
      uri:
        "https://templates.accordproject.org/archives/car-rental-tr@0.10.1.cta",
      name: "Car Rental (TR)",
      version: "^0.20.10",
      description: "Ta Simple Car Rental Contract in Turkish Language",
      itemType: 'template'
    },
    {
      uri:
        "https://templates.accordproject.org/archives/certificate-of-incorporation@0.3.1.cta",
      name: "Certificate Of Incorporation",
      version: "^0.20.10",
      description: "This is a sample Certificate of Incorporation.",
      itemType: 'template'
    }
  ];
  const mockImport = () => {
    alert(
      "A function callback which is called when a link to import a new library item is clicked"
    );
  };
  const mockUpload = () => {
    alert(
      "A function callback which is called when a link to upload a new library item is clicked"
    );
  };
  const mockAddItem = () => {
    alert(
      "A function callback which is called when a link to add a new library item is clicked"
    );
  };
  const mockPrimaryButtonClick = () => {
    alert(
      "A function callback which is called when a primary button in a library item card is clicked."
    );
  };
  const mockSecondaryButtonClick = () => {
    alert(
      "A function callback which is called when a secondary button in a library item card is clicked."
    );
  };
  return (
    <Library
      items={items}
      onPrimaryButtonClick={mockPrimaryButtonClick}
      onSecondaryButtonClick={mockSecondaryButtonClick}
      onUploadItem={mockUpload}
      onImportItem={mockImport}
      onAddItem={mockAddItem}
    />
  );
};

library.story = {
  component: library,
  decorators: [withA11y, withKnobs],
  parameters: {
    notes: docs
  }
};
