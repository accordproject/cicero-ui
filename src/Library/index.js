import _ from 'lodash';

/* React */
import React, { useState, useCallback } from 'react';
import PropTypes from 'prop-types';

/* Styling */
import styled from 'styled-components';
import { Input, Checkbox } from 'semantic-ui-react';

/* Internal */
import isQueryMatch from '../utilities/isQueryMatch';
import LibraryItemCard from './Components/LibraryItemCard';
import { ImportComponent, UploadComponent, NewItemComponent } from './Buttons';

const Wrapper = styled.div`
  font-family: 'IBM Plex Sans', sans-serif;
  position: relative;
  max-width: 355px;
  height: inherit;
  grid-template-areas:  "header" "functionTemps";
  grid-template-rows:    55px auto;
  grid-template-columns: 100%;
`;

const Header = styled.div`
  position: relative;
  font-family: 'IBM Plex Sans', sans-serif;
  font-weight: 800;
  font-size: 1em;
  margin-bottom: 10px;

  display: grid;
  grid-area: header;
  grid-template-columns: 1fr 1fr;
  grid-template-areas: "title imports";
`;

const HeaderImports = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
`;

const Functionality = styled.div`
  font-family: 'IBM Plex Sans', sans-serif;
  max-width: 430px;
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  grid-area: functionTemps;
  display: grid;
`;

const ItemTypeFilterContainer = styled.div`
  display: flex;
  margin-bottom: 15px;
  margin-top: 15px;
`;

const searchInputStyles = {
  margin: '5px 0',
  width: '100%',
  maxHeight: '40px',
  borderRadius: '3px',
  boxShadow: 'inset 0 0 4px 0 #ABABAB',
};

const LibraryItemCards = styled.div`
  margin: 0;
  width: 100%;
  height: calc(100% - 54px);
  display: inherit;
  overflow-y: scroll;
`;

/**
 * A Library component that displays the filtered list of library items
 */

const LibraryComponent = (props) => {
  const [query, setQuery] = useState([]);
  const [itemTypeFilter, setItemTypeFilter] =
    useState(props.itemTypes.reduce((acc, el) => ({ ...acc, [el.type]: true }), {}));

  const onQueryChange = (e, input) => {
    const inputQuery = input.value.toLowerCase().trim().split(' ').filter(q => q.length);
    if (inputQuery !== query) {
      setQuery(inputQuery);
    }
  };

  const renderHeader = useCallback(() => {
    if (!props.onUploadItem && !props.onImportItem) return null;

    return(
      <Header>
        <HeaderImports>
          <ImportComponent onImportItem={props.onImportItem} />
          <UploadComponent onUploadItem={props.onUploadItem} />
        </HeaderImports>
      </Header>
    );
  }, [props.onImportItem, props.onUploadItem]);

  const filterItemsByQuery = useCallback(() => {
    let filtered = props.items;

    if (query.length) {
      filtered = _.filter(
        filtered,
        i => isQueryMatch([i.name, i.displayName, i.uri], query)
      );
    }

    return filtered;
  }, [props.items, query]);

  const filterItems = useCallback(() => {
    let filtered = filterItemsByQuery();

    const currentFilter = Object.keys(_.pickBy(itemTypeFilter));
    if (currentFilter.length !== props.itemTypes.length) {
      filtered = filtered.filter(item => currentFilter.indexOf(item.itemType) > -1);
    }

    return filtered;
  }, [query, props.items, itemTypeFilter]);

  const renderItemTypeFilter = useCallback(() => {
    if (props.itemTypes.length === 1) return null;

    return(
      <ItemTypeFilterContainer>
        {
          props.itemTypes.map((itemType, idx) => (
            <Checkbox
              key={idx}
              label={itemType.filterName}
              style={{ marginRight: '10px' }}
              onChange={() => setItemTypeFilter({
                ...itemTypeFilter,
                [itemType.type]: !itemTypeFilter[itemType.type],
              })}
              checked={itemTypeFilter[itemType.type]}
            />
          ))
        }
      </ItemTypeFilterContainer>
    );
  }, [itemTypeFilter]);

  const filtered = filterItems();
  const dislayedItemTypes =
    props.itemTypes.reduce((acc, el) => ({ ...acc, [el.type]: el.name }), {});

  return (
    <Wrapper>
      { renderHeader() }
      <Functionality>
        { renderItemTypeFilter() }
        <Input
          className="cicero-ui__library-search-input"
          fluid
          icon="search"
          placeholder="Search..."
          onChange={onQueryChange}
          style={searchInputStyles}
        />
        <NewItemComponent onAddItem={props.onAddItem} />
      </Functionality>
      <LibraryItemCards>
        {
          filtered.length
            ? filtered.map(item => (
                <LibraryItemCard
                  key={item.uri}
                  item={item}
                  itemTypeName={dislayedItemTypes[item.itemType] || 'ITEM'}
                  onPrimaryButtonClick={props.onPrimaryButtonClick}
                  onSecondaryButtonClick={props.onSecondaryButtonClick}
                />
              ))
            : <p style={{ textAlign: 'center' }}>No results found</p>
        }
      </LibraryItemCards>
    </Wrapper>
  );
};

LibraryComponent.propTypes = {
  onUploadItem: PropTypes.func,
  onImportItem: PropTypes.func,
  onAddItem: PropTypes.func,
  onPrimaryButtonClick: PropTypes.func.isRequired,
  onSecondaryButtonClick: PropTypes.func.isRequired,
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  itemTypes: PropTypes.arrayOf(PropTypes.object).isRequired,
};

LibraryComponent.defaultProps = {
  itemTypes: [{ name: 'CLAUSE TEMPLATE', type: 'template', filterName: 'Templates' }]
};

export default LibraryComponent;
