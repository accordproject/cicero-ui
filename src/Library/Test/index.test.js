import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import LibraryItemCard from '../Components/LibraryItemCard';
import Library from '../index';

const libraryItemsArray = [
  {
    description: 'This clause allows the receiver of goods to inspect them for a given time period after delivery.',
    name: 'acceptance-of-delivery',
    uri: 'ap://acceptance-of-delivery@0.11.0#311de48109cce10e6b2e33ef183ccce121886d0b76754d649d5054d1084f93cd',
    version: '0.11.0'
  },
  {
    description: 'a Simple Car Rental Contract in Turkish Language',
    name: 'car-rental-tr',
    uri: 'ap://car-rental-tr@0.8.0#1d6de2328745fe4bae726e0fb887fee6d3fbd00b3d717e30e38dc7bffd7222fa',
    version: '0.8.0'
  },
  {
    description: 'This clause is a copyright license agreement.',
    name: 'copyright-license',
    uri: 'ap://copyright-license@0.12.0#55d73ab5dacc642d8a7cd0bc95c7da5580bd031e00b955616837abd50be08b6b',
    version: '0.12.0'
  },
  {
    description: 'A sample demandforecast clause.',
    name: 'demandforecast',
    uri: 'ap://demandforecast@0.11.0#ac252106feb1f685f5ecceb12b967bb8210ca5283660b5bf72aeb5b397342f87',
    version: '0.11.0'
  },
  {
    description: 'Counts events from DocuSign connect with a given envelope status.',
    name: 'docusign-connect',
    uri: 'ap://docusign-connect@0.5.0#4b44811f4f3e4899c026c35f4180d6c555f9101e8fcca35a13ad09aa2e28b7f7',
    version: '0.5.0'
  },
  {
    description: 'This is a clause enforcing healthy eating habits in employees.',
    name: 'eat-apples',
    uri: 'ap://eat-apples@0.8.0#5d04f0bb8698e1e7565e496aa97beb055b1f3e7fb353d0b0447aeaac59729ed2',
    version: '0.8.0'
  },
];

const mockUpload = jest.fn();
const mockImport = jest.fn();
const addNewTemplate = jest.fn();
const mockPrimaryButtonClick = jest.fn();
const mockSecondaryButtonClick = jest.fn();

const propInput = {
  items: libraryItemsArray,
  onUploadItem: mockUpload,
  onImportItem: mockImport,
  onAddItem: addNewTemplate,
  onPrimaryButtonClick: mockPrimaryButtonClick,
  onSecondaryButtonClick: mockSecondaryButtonClick,
};

describe('<Library />', () => {
  describe('on initialization', () => {
    it('renders page correctly', () => {
      const component = shallow(<Library {...propInput} />);
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('renders conditional buttons', () => {
    it('with functions passed in', () => {
      const component = shallow(<Library {...propInput} />);
      expect(component.find('UploadComponent')).toHaveLength(1);
      expect(component.find('ImportComponent')).toHaveLength(1);
      expect(component.find('NewItemComponent')).toHaveLength(1);
    });

    it('without functions passed in', () => {
      const component = shallow(<Library
        items={libraryItemsArray}
        onPrimaryButtonClick={mockPrimaryButtonClick}
        onSecondaryButtonClick={mockSecondaryButtonClick}
      />);
      expect(component.find('UploadComponent')).toHaveLength(0);
      expect(component.find('ImportComponent')).toHaveLength(0);
      expect(component.find('NewItemComponent')).toHaveLength(1);
    });
  });

  describe('runs functions passed into it', () => {
    it('add To Contract function runs', () => {
      const component = shallow(
        <LibraryItemCard
          item={libraryItemsArray[0]}
          onPrimaryButtonClick={mockPrimaryButtonClick}
          onSecondaryButtonClick={mockSecondaryButtonClick}
        />
      );
      expect(component.find('CardActions').prop('onPrimaryButtonClick')).toEqual(mockPrimaryButtonClick);
    });
  });
});
