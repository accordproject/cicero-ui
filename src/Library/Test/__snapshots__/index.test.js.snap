// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Library /> on initialization renders page correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <ImportComponent
        onImportItem={[MockFunction]}
      />
      <UploadComponent
        onUploadItem={[MockFunction]}
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <Input
      className="cicero-ui__library-search-input"
      fluid={true}
      icon="search"
      onChange={[Function]}
      placeholder="Search..."
      style={
        Object {
          "borderRadius": "3px",
          "boxShadow": "inset 0 0 4px 0 #ABABAB",
          "margin": "5px 0",
          "maxHeight": "40px",
          "width": "100%",
        }
      }
      type="text"
    />
    <NewItemComponent
      onAddItem={[MockFunction]}
    />
  </styled.div>
  <styled.div>
    <LibraryItemCard
      item={
        Object {
          "description": "This clause allows the receiver of goods to inspect them for a given time period after delivery.",
          "name": "acceptance-of-delivery",
          "uri": "ap://acceptance-of-delivery@0.11.0#311de48109cce10e6b2e33ef183ccce121886d0b76754d649d5054d1084f93cd",
          "version": "0.11.0",
        }
      }
      itemTypeName="ITEM"
      key="ap://acceptance-of-delivery@0.11.0#311de48109cce10e6b2e33ef183ccce121886d0b76754d649d5054d1084f93cd"
      onPrimaryButtonClick={[MockFunction]}
      onSecondaryButtonClick={[MockFunction]}
    />
    <LibraryItemCard
      item={
        Object {
          "description": "a Simple Car Rental Contract in Turkish Language",
          "name": "car-rental-tr",
          "uri": "ap://car-rental-tr@0.8.0#1d6de2328745fe4bae726e0fb887fee6d3fbd00b3d717e30e38dc7bffd7222fa",
          "version": "0.8.0",
        }
      }
      itemTypeName="ITEM"
      key="ap://car-rental-tr@0.8.0#1d6de2328745fe4bae726e0fb887fee6d3fbd00b3d717e30e38dc7bffd7222fa"
      onPrimaryButtonClick={[MockFunction]}
      onSecondaryButtonClick={[MockFunction]}
    />
    <LibraryItemCard
      item={
        Object {
          "description": "This clause is a copyright license agreement.",
          "name": "copyright-license",
          "uri": "ap://copyright-license@0.12.0#55d73ab5dacc642d8a7cd0bc95c7da5580bd031e00b955616837abd50be08b6b",
          "version": "0.12.0",
        }
      }
      itemTypeName="ITEM"
      key="ap://copyright-license@0.12.0#55d73ab5dacc642d8a7cd0bc95c7da5580bd031e00b955616837abd50be08b6b"
      onPrimaryButtonClick={[MockFunction]}
      onSecondaryButtonClick={[MockFunction]}
    />
    <LibraryItemCard
      item={
        Object {
          "description": "A sample demandforecast clause.",
          "name": "demandforecast",
          "uri": "ap://demandforecast@0.11.0#ac252106feb1f685f5ecceb12b967bb8210ca5283660b5bf72aeb5b397342f87",
          "version": "0.11.0",
        }
      }
      itemTypeName="ITEM"
      key="ap://demandforecast@0.11.0#ac252106feb1f685f5ecceb12b967bb8210ca5283660b5bf72aeb5b397342f87"
      onPrimaryButtonClick={[MockFunction]}
      onSecondaryButtonClick={[MockFunction]}
    />
    <LibraryItemCard
      item={
        Object {
          "description": "Counts events from DocuSign connect with a given envelope status.",
          "name": "docusign-connect",
          "uri": "ap://docusign-connect@0.5.0#4b44811f4f3e4899c026c35f4180d6c555f9101e8fcca35a13ad09aa2e28b7f7",
          "version": "0.5.0",
        }
      }
      itemTypeName="ITEM"
      key="ap://docusign-connect@0.5.0#4b44811f4f3e4899c026c35f4180d6c555f9101e8fcca35a13ad09aa2e28b7f7"
      onPrimaryButtonClick={[MockFunction]}
      onSecondaryButtonClick={[MockFunction]}
    />
    <LibraryItemCard
      item={
        Object {
          "description": "This is a clause enforcing healthy eating habits in employees.",
          "name": "eat-apples",
          "uri": "ap://eat-apples@0.8.0#5d04f0bb8698e1e7565e496aa97beb055b1f3e7fb353d0b0447aeaac59729ed2",
          "version": "0.8.0",
        }
      }
      itemTypeName="ITEM"
      key="ap://eat-apples@0.8.0#5d04f0bb8698e1e7565e496aa97beb055b1f3e7fb353d0b0447aeaac59729ed2"
      onPrimaryButtonClick={[MockFunction]}
      onSecondaryButtonClick={[MockFunction]}
    />
  </styled.div>
</styled.div>
`;
